crates/rooch-relayer/src/actor/relayer.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+use super::ethereum_relayer::EthereumRelayer;
+use super::messages::RelayTick;
+use crate::{Relayer, TxSubmiter};
+use anyhow::Result;
+use async_trait::async_trait;
+use coerce::actor::{context::ActorContext, message::Handler, Actor};
+use moveos_types::{gas_config::GasConfig, transaction::MoveAction};
+use rooch_rpc_api::jsonrpc_types::KeptVMStatusView;
+use rooch_rpc_client::ClientBuilder;
+use rooch_types::{
+    address::RoochAddress,
+    crypto::RoochKeyPair,
+    transaction::{rooch::RoochTransactionData, AbstractTransaction},
+};
+use tracing::{info, warn};
+
+pub struct RelayerActor {
+    chain_id: u64,
+    relayer_address: RoochAddress,
+    max_gas_amount: u64,
+    relayer_key: RoochKeyPair,
+    tx_submiter: Box<dyn TxSubmiter>,
+    relayers: Vec<Box<dyn Relayer>>,
+}
+
+impl RelayerActor {
+    /// Create a new RelayerActor, use rooch_rpc_client::Client as TxSubmiter
+    pub async fn new_for_client(
+        relayer_key: RoochKeyPair,
+        eth_rpc_url: &str,
+        rooch_rpc_url: &str,
+    ) -> Result<Self> {
+        let rooch_rpc_client = ClientBuilder::default().build(rooch_rpc_url).await?;
+        Self::new(relayer_key, eth_rpc_url, rooch_rpc_client).await
+    }
+
+    pub async fn new<T: TxSubmiter + 'static>(
+        relayer_key: RoochKeyPair,
+        eth_rpc_url: &str,
+        tx_submiter: T,
+    ) -> Result<Self> {
+        let chain_id = tx_submiter.get_chain_id().await?;
+        let relayer_address = relayer_key.public().address();
+        let eth_relayer = EthereumRelayer::new(eth_rpc_url)?;
+        let relayers: Vec<Box<dyn Relayer>> = vec![Box::new(eth_relayer)];
+        Ok(Self {
+            chain_id,
+            relayer_address,
+            max_gas_amount: GasConfig::DEFAULT_MAX_GAS_AMOUNT,
+            relayer_key,
+            relayers,
+            tx_submiter: Box::new(tx_submiter),
+        })
+    }
+
+    async fn tick(&mut self) -> Result<()> {
```

Comment:
L1 information could be delayed by several block confirmations to minimize the impact of reorgs.

For eth could be ~24seconds (2 confirmations) just like what optimism did

Reply:
I plan to implement this logic in the contract, The eth light client contract should maintain the status of the chain, and decide the finality.

Reply:
It's better this way